### DISCUSSION TITLE
Return type of item id from session.exec is `int | None`. How can I make it `int`.

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi/sqlmodel#1486] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlalchemy.exc import NoResultFound
from sqlmodel import Field, Session, SQLModel, create_engine

sqlite_file_name = "database.db"
sqlite_url = f"sqlite:///{sqlite_file_name}"

engine = create_engine(sqlite_url, echo=True)


def create_db_and_tables():
    SQLModel.metadata.create_all(engine)


class Person(SQLModel):
    id: int | None = Field(default=None, primary_key=True, index=True)
    name: str


def add_person():
    with Session(engine) as session:
        john = Person(name="John")
        session.add(john)
        session.commit()


def get_person_or_404(id: int) -> Person:
    with Session(engine) as session:
        person = session.get(Person, id)

        if not person:
            raise NoResultFound("Person not found")

        return person


def main():
    create_db_and_tables()
    add_person()
    person = get_person_or_404(1)
    print(f"person.id: {person.id}")


if __name__ == "__main__":
    main()
```

### Description

If get person from db and db always returns the person with the id (type: `int`). However person returned by `person = get_person_or_404(1)` has person with id optional.

Hence, the type of `person.id` is `int | None` even though it is returned from DB.

How I type the `get_person_or_404` function to let it know that id from `Person` class is `int` and not `int | None`.

Thanks.

### Operating System

Linux

### Operating System Details

Ubuntu 21.10

### SQLModel Version

0.0.6

### Python Version

3.10.2

### Additional Context

![Screenshot from 2022-03-24 13-38-18](https://user-images.githubusercontent.com/47495003/159873502-5b8c28cb-8b53-420c-9d1b-8061f05f73b8.png)

**Comment by jd-solanki** [comment]
As of now, I am creating a new class and `cast` the return value:

```python
def get_product(session: Session, id: PositiveInt) -> ProductDbRead | None:
    return cast(ProductDbRead, session.get(Product, id))
```

**Comment by YuriiMotov** [comment, marked as answer]
Yes, creating a separate model with required `id` field is a conventional approach.

```py
class PersonBase(SQLModel):
    name: str

class Person(PersonBase):
    id: int | None = Field(default=None, primary_key=True, index=True)

class PersonRead(PersonBase):
    id: int

...

def get_person_or_404(id: int) -> PersonRead:
    with Session(engine) as session:
        person = session.get(Person, id)

        if not person:
            raise NoResultFound("Person not found")

        return PersonRead.model_validate(person, from_attributes=True)
```


This is explained here: https://sqlmodel.tiangolo.com/tutorial/fastapi/multiple-models/
More specifically: https://sqlmodel.tiangolo.com/tutorial/fastapi/multiple-models/#the-heropublic-data-model
